pkg/util/vmopv1/hardware.go

```diff
@@ -96,69 +96,65 @@ func GetControllerSharingMode(
 // controller type, bus number, and sharing mode.
 // For SCSI controller type, the type is ParaVirtualSCSI by default.
 func CreateNewController(
-	controllerID pkgutil.ControllerID,
+	controllerType vmopv1.VirtualControllerType,
+	busNumber int32,
 	sharingMode vmopv1.VirtualControllerSharingMode,
 ) ControllerSpec {
-	switch controllerID.ControllerType {
+	switch controllerType {
 	case vmopv1.VirtualControllerTypeSCSI:
 		return vmopv1.SCSIControllerSpec{
-			BusNumber:   controllerID.BusNumber,
+			BusNumber:   busNumber,
 			Type:        vmopv1.SCSIControllerTypeParaVirtualSCSI,
 			SharingMode: sharingMode,
 		}
 	case vmopv1.VirtualControllerTypeSATA:
 		return vmopv1.SATAControllerSpec{
-			BusNumber: controllerID.BusNumber,
+			BusNumber: busNumber,
 		}
 	case vmopv1.VirtualControllerTypeNVME:
 		return vmopv1.NVMEControllerSpec{
-			BusNumber:   controllerID.BusNumber,
+			BusNumber:   busNumber,
 			SharingMode: sharingMode,
 		}
 	case vmopv1.VirtualControllerTypeIDE:
 		return vmopv1.IDEControllerSpec{
-			BusNumber: controllerID.BusNumber,
+			BusNumber: busNumber,
 		}
 	default:
 		return nil
 	}
 }
 
-// BuildVMControllersMap builds a map of controller ID to controller specification
-// from the specified VM's spec.hardware.controllers.
+// BuildVMControllersMap builds a map of controller type to a map of bus number
```

Comment:
Return a struct that returns multiple maps with a helper function. I generally do not like functions called "Build" or "Get" map. That's a private function. If public there should be more first-class nature to it. See https://github.com/vmware-tanzu/vm-operator/pull/1253/files#diff-8ffd15b3281f975f52b08d5a244f1bd4a5fc3ad4f5517972ee4f6cf011ea0684R109-R257 for an example of what I mean.